google-cloud-firestore/src/main/java/com/google/cloud/firestore/Watch.java

```diff
@@ -337,15 +337,18 @@ private void closeStream(final Throwable throwable) {
           new Runnable() {
             @Override
             public void run() {
-              listener.onEvent(
-                  null,
-                  throwable instanceof FirestoreException
-                      ? (FirestoreException) throwable
-                      : FirestoreException.forApiException(
-                          new ApiException(
-                              throwable,
-                              GrpcStatusCode.of(getStatus(throwable).getCode()),
-                              false)));
+              if (throwable instanceof FirestoreException) {
```

Comment:
These changes are necessary becuse of grpc interrupting the channel while this could still be listenting?

Reply:
Yes, and the exceptions thrown then do no have status codes. This is all based on local debugging.